task1/index.js

```diff
@@ -0,0 +1,117 @@
+const notes = [
+    {
+        id: 1,
+        title: "Recipe",
+        description: "Ingredients include 2 eggs...",
+        pagesCount: 2,
+        isMarked: false,
+        access: []
+    },
+    {
+        id: 2,
+        title: "Workouts",
+        description: "3 sets of squats...",
+        pagesCount: 1,
+        isMarked: true,
+        access: []
+    },
+    {
+        id: 3,
+        title: "Passwords",
+        description: "VISA ...",
+        pagesCount: 6,
+        isMarked: true,
+        access: []
+    },
+    {
+        id: 4,
+        title: "To Do 2021",
+        description: "1. Learn JS...",
+        pagesCount: 3,
+        isMarked: false,
+        access: []
+    }
+]
+
+// MAP POLYFIL
+Array.prototype.myMap = function (callback) {
+    let result = []
+    for (let i = 0; i < this.length; i++) {
+        result.push(callback(this[i], i, this))
+    }
+    return result
+}
+
+
+// MAP
+Array.prototype.myMap = function (callback) {
+    let result = []
+    for (let i = 0; i < this.length; i++) {
+        result.push(callback(this[i], i, this))
+    }
+    return result
+}
+
+let result = notes.myMap((item) => {
+    return item.id + ' ' + item.title
+})
+console.log(result);
+
+
+// FILTER POLYFIL
+Array.prototype.myFilter = function (callback) {
+    let result = []
+    for (let i = 0; i < this.length; i++) {
+       if(callback(this[i], i, this)) {
+            result.push(this[i])
+       }
+    }
+    return result
+}
+
+
+// FILTER
+Array.prototype.myFilter = function (callback) {
+    let result = []
+    for (let i = 0; i < this.length; i++) {
+       if(callback(this[i], i, this)) {
+            result.push(this[i])
+       }
+    }
+    return result
+}
+
+let result = notes.myFilter((item) => {
+    return item.isMarked === true 
+})
```

Comment:
There is no need to write `return` statement!
There is no need to check `true`!
```suggestion
const filteredNotes = notes.myFilter((item) => note.isMarked)
```